src/services/formatting/formatting.ts

```diff
@@ -720,6 +720,9 @@ namespace ts.formatting {
                     // proceed any parent tokens that are located prior to child.getStart()
                     const tokenInfo = formattingScanner.readTokenInfo(node);
                     if (tokenInfo.token.end > childStartPos) {
+                        if (tokenInfo.token.pos > childStartPos) {
+                            formattingScanner.skipToStartOf(child);
+                        }

```

Comment:
if the outer token's start position is ahead of the childs start, then we're basically in a weird state - try reparsing